contract/src/main.rs

```diff
@@ -914,54 +1008,47 @@ pub extern "C" fn set_approval_for_all() {
     )
     .unwrap_or_revert();
 
-    let operators_seed_uref = utils::get_uref(
-        OPERATOR,
-        NFTCoreError::MissingStorageUref,
-        NFTCoreError::InvalidStorageUref,
-    );
-
-    let callers_owned_tokens = match utils::get_reporting_mode() {
-        OwnerReverseLookupMode::NoLookUp => utils::get_owned_token_ids_by_token_number(),
-        OwnerReverseLookupMode::Complete => utils::get_owned_token_ids_by_page(),
-    };
+    // If caller tries to approve itself as operator that's probably a mistake and we revert.
+    if caller == operator {
+        runtime::revert(NFTCoreError::InvalidAccount);
+    }
 
     // Depending on approve_all we either approve all or disapprove all.
-    let operator = if approve_all { Some(operator) } else { None };
-    for token_id in &callers_owned_tokens {
-        // We assume a burned token cannot be approved
-        if utils::is_token_burned(token_id) {
-            continue;
-        }
+    let operator: Option<Key> = if approve_all { Some(operator) } else { None };
+    let owner_item_key = utils::get_stringified_token_owner_item_key(caller);
 
-        storage::dictionary_put(
-            operators_seed_uref,
-            &token_id.get_dictionary_item_key(),
-            operator,
-        );
-    }
+    upsert_dictionary_value_from_key(OPERATORS, &owner_item_key, operator);
+
+    let events_mode: EventsMode = utils::get_stored_value_with_user_errors::<u8>(
+        EVENTS_MODE,
+        NFTCoreError::MissingEventsMode,
+        NFTCoreError::InvalidEventsMode,
+    )
+    .try_into()
+    .unwrap_or_revert();
 
     match events_mode {
         EventsMode::NoEvents => {}
         EventsMode::CES => {
-            casper_event_standard::emit(ApprovalForAll::new(
-                token_owner,
-                operator,
-                callers_owned_tokens,
-            ));
+            casper_event_standard::emit(ApprovalForAll::new(caller, operator));
         }
         EventsMode::CEP47 => {
-            for callers_owned_token in callers_owned_tokens {
+            let caller_owned_tokens = match utils::get_reporting_mode() {
+                OwnerReverseLookupMode::NoLookUp => utils::get_owned_token_ids_by_token_number(),
+                OwnerReverseLookupMode::Complete => utils::get_owned_token_ids_by_page(),
+            };
+            for callers_owned_token in caller_owned_tokens {
```

Comment:
I think here we should also have the `ApprovalForAll` event. The CEP-47 event mode refers to CEP-47 standard event implementation, and to the standard itself.

Reply:
Hi, I committed this change however I am checking with the rest of the team is this is really what was intended. 

https://github.com/casper-ecosystem/cep-78-enhanced-nft/pull/181/commits/2eee528a7e30d8b44e9451f9abd722ea9b75dfc3